backend/attendance-service/src/main/java/org/egov/service/StaffService.java

```diff
@@ -56,22 +65,21 @@ public StaffPermissionRequest createAttendanceStaff(StaffPermissionRequest staff
         staffServiceValidator.validateStaffPermissionRequestParameters(staffPermissionRequest);
 
         //extract registerIds and staffUserIds from client request
-        List<String> staffIds=new ArrayList<>();
-        List<String> registerIds=new ArrayList<>();
-        List<StaffPermission> staffPermissionListFromRequest=staffPermissionRequest.getStaffPermissionList();
-        String rootTenantId=staffPermissionRequest.getStaffPermissionList().get(0).getTenantId();
-        for(StaffPermission staffPermission:staffPermissionListFromRequest){
-            staffIds.add(staffPermission.getUserId());
-            registerIds.add(staffPermission.getRegisterId());
-        }
+        String rootTenantId = staffPermissionRequest.getStaff().get(0).getTenantId();
```

Comment:
You can't assume that it would be the `rootTenantId`. 
For this particular case, I can say this for sure wouldn't be the root tenant id.

Reply:
`//incoming createRequest validation
        staffServiceValidator.validateStaffPermissionRequestParameters(staffPermissionRequest);

        //extract registerIds and staffUserIds from client request
        String rootTenantId = staffPermissionRequest.getStaff().get(0).getTenantId();`

validateStaffPermissionRequestParameters checks if all staff from request have the same tenantIds. After that check I am assuming rootTenantd from the first staff object. Hence we can be sure that the rootTenantId is valid

Reply:
Variable name changed to tenantId as per discussion